typescript/src/entity.ts

```diff
@@ -437,7 +449,7 @@ export class Entity {
   }
 
   public save = () => {
-    const primaryKey: number = (this as any).id;
+    const primaryKey: number = this.getPrimaryKeyValue();
```

Comment:
Do not sure about how typescript works, for now I can use my knowledge about JAVA to infer it.

just the  same question as above, primaryKeyValue can be string for token